megamek/src/megamek/client/ui/swing/util/ScaledImageFileFactory.java

```diff
@@ -0,0 +1,215 @@
+     *            produce the item. This value must not be <code>null</code>.
+     * @return an <code>ItemFile</code> for the given file.
+     * @throws <code>IllegalArgumentException</code> if the <code>file</code>
+     *             is <code>null</code>.
+     */
+    public ItemFile getItemFile(final File file)
+            throws IllegalArgumentException {
+
+        // Validate the input.
+        if (null == file) {
+            throw new IllegalArgumentException("A null image file was passed."); //$NON-NLS-1$
+        }
+
+        // Construct an anonymous class that gets an Image for the file.
+        return new ItemFile() {
+
+            private File itemFile = file; // copy the file entry
+            private Image image = null; // cache the Image
+
+            public Object getItem() throws Exception {
+                // Cache the image on first use.
+                if (null == image) {
+                    String name = itemFile.getAbsolutePath();
+                    image = ImageUtil.loadImageFromFile(name);
+                }
+                // Return a copy of the image.
+                return ImageUtil.getScaledImage(image, 84, 72);
+            } // End getItem()
+        };
+
+    }
+
+    /**
+     * Get the <code>ItemFile</code> for the given <code>ZipEntry</code> in
+     * the <code>ZipFile</code>.
+     * 
+     * @param zipEntry - the <code>ZipEntry</code> that will be read to
+     *            produce the item. This value must not be <code>null</code>.
+     * @param zipFile - the <code>ZipFile</code> object that contains the
+     *            <code>ZipEntry</code> that will produce the item. This value
+     *            must not be <code>null</code>.
+     * @return an <code>ItemFile</code> for the given zip file entry.
+     * @throws <code>IllegalArgumentException</code> if either the
+     *             <code>zipEntry</code> or the <code>zipFile</code> is
+     *             <code>null</code>.
+     */
+    public ItemFile getItemFile(final ZipEntry zipEntry, final ZipFile zipFile)
+            throws IllegalArgumentException {
+
+        // Validate the input.
+        if (null == zipEntry) {
+            throw new IllegalArgumentException("A null ZIP entry was passed."); //$NON-NLS-1$
+        }
+        if (null == zipFile) {
+            throw new IllegalArgumentException("A null ZIP file was passed."); //$NON-NLS-1$
+        }
+
+        // Construct an anonymous class that gets an Image for the file.
+        return new ItemFile() {
+
+            private ZipEntry itemEntry = zipEntry; // copy the ZipEntry
+            private Image image = null; // cache the Image
+
+            public Object getItem() throws Exception {
+
+                // Cache the image on first use.
+                if (null == image) {
+
+                    // Get ready to read from the item.
+                    InputStream in = new BufferedInputStream(zipFile
```

Comment:
According to LGTM (and it seems to be right), you should close this input stream regardless of whether the call succeeds or fails.